src/leaflike/bookmarks.clj

```diff
@@ -191,19 +223,23 @@
 (defn edit-view
   [{:keys [params] :as request}]
   (let [username (get-in request [:session :username])
-        user      (hutils/get-user request)
+        user (hutils/get-user request)
         bookmark-id (Integer/parseInt (:bookmark-id params))
-        bookmark (bm-db/fetch-bookmark bookmark-id (:id user))
+        bookmark (bm-db/fetch-bookmark bookmark-id)
         error-msg (if bookmark
                     (get-in request [:flash :error-msg])
                     "The bookmark you're trying to edit does not exist.")
-        all-tags (map :name (tags-db/fetch-tags {:user-id (:id user)}))]
-    (-> (res/response (layout/user-view "Edit Bookmark"
-                                        username
-                                        (views/bookmark-form anti-forgery/*anti-forgery-token*
-                                                             "/bookmarks/edit"
-                                                             (assoc bookmark
-                                                                    :all-tags all-tags))
+        all-tags (map :name (tags-db/fetch-tags {:user-id (:id user)}))
+        collaborator-ids (map :user_id (bm-db/get-collaborator-ids bookmark-id))]
+    (-> (res/response (layout/user-view "Edit Bookmark" username (views/bookmark-form
+                                          anti-forgery/*anti-forgery-token*
+                                          "/bookmarks/edit"
+                                          (assoc bookmark
+                                            :all-tags all-tags
+                                            :collaborators
+                                            (map :username
+                                                 (if (not-empty collaborator-ids)
+                                                   (bm-db/get-collaborators-from-ids collaborator-ids)))))
```

Comment:
we don't need two different queries to first get collaborator-ids and get their names. you can fetch the list of collaborators and map over them to get the usernames (no `if` necessary either because map over an empty list is valid and gives you an empty list).

Reply:
I fetch the collaborator IDs from the `bookmark_user` table (which doesn't consist usernames) and then use the retrieved IDs to fetch the usernames from the users table. I could also do a join of users and bookmark_user to achieve this at once. 

Regarding the approach you've suggested: Is this using `fmap`?